### ISSUE TITLE
Not efficient on large datasets or working with databases

### ISSUE THREAD

**Issue by Nixellion** [SergioLlana/datatables-flask-serverside#9]
I am currently trying to adapt this to working with an sqlite database using peewee ORM. First thing I did was trying to use it without altering the code, just dumping the whole database with about 1000 entries into it. It sort of worked, but it takes about 10 seconds to process and show data on page, which is obviously not acceptable if it only takes <1 second to query just 10 of the required entries and send only them.

So I don't see how this particular module is useful as is. The point of server side processing, even if you work with large dataset already in RAM (which uses who?) is to speed things up. In this example when you have like 10-100 rows you might as well just dump them all to the client side and let them sort it there.

As an example it works great, though, but working on rewriting it to work with peewee ORM natively

**Comment by rinaldipratama** [comment]
> I am currently trying to adapt this to working with an sqlite database using peewee ORM. First thing I did was trying to use it without altering the code, just dumping the whole database with about 1000 entries into it. It sort of worked, but it takes about 10 seconds to process and show data on page, which is obviously not acceptable if it only takes <1 second to query just 10 of the required entries and send only them.
> 
> So I don't see how this particular module is useful as is. The point of server side processing, even if you work with large dataset already in RAM (which uses who?) is to speed things up. In this example when you have like 10-100 rows you might as well just dump them all to the client side and let them sort it there.
> 
> As an example it works great, though, but working on rewriting it to work with peewee ORM natively

You're right, I've tried this but when working with 100.000++ data it will slowly respond (about 15 seconds). I think it's better to modify this code with support for SQL/SQLite query like datatables ssp in php
